cmd/gapir/cc/main.cpp

```diff
@@ -76,6 +77,235 @@ std::vector<uint32_t> memorySizes {
+                                     // flag is not supported.
+        opts.SetMode(kReplayServer);
+        opts.onDiskCacheOptions.enabled = true;
+      } else if (strcmp(argv[i], "--disk-cache-path") == 0) {
+        ensureNotAndroid(
+            "--disk-cache-path");  // Fatal error if running on android. This
+                                   // flag is not supported.
+        opts.SetMode(kReplayServer);
+        if (i + 1 >= argc) {
+          GAPID_FATAL("Usage: --disk-cache-path <cache-directory>");
+        }
+        opts.onDiskCacheOptions.path = argv[++i];
+      } else if (strcmp(argv[i], "--cleanup-on-disk-cache") == 0) {
+        ensureNotAndroid(
+            "--cleanup-on-disk-cache");  // Fatal error if running on android.
+                                         // This flag is not supported.
+        opts.onDiskCacheOptions.cleanUp = true;
+      } else if (strcmp(argv[i], "--port") == 0) {  // check it works
+        opts.SetMode(kReplayServer);
+        if (i + 1 >= argc) {
+          GAPID_FATAL("Usage: --port <port_num>");
+        }
+        opts.portArgStr = argv[++i];
+      } else if (strcmp(argv[i], "--log-level") == 0) {
+        if (i + 1 >= argc) {
+          GAPID_FATAL("Usage: --log-level <F|E|W|I|D|V>");
+        }
+        switch (argv[++i][0]) {
+          case 'F':
+            opts.logLevel = LOG_LEVEL_FATAL;
+            break;
+          case 'E':
+            opts.logLevel = LOG_LEVEL_ERROR;
+            break;
+          case 'W':
+            opts.logLevel = LOG_LEVEL_WARNING;
+            break;
+          case 'I':
+            opts.logLevel = LOG_LEVEL_INFO;
+            break;
+          case 'D':
+            opts.logLevel = LOG_LEVEL_DEBUG;
+            break;
+          case 'V':
+            opts.logLevel = LOG_LEVEL_VERBOSE;
+            break;
+          default:
+            GAPID_FATAL("Usage: --log-level <F|E|W|I|D|V>");
+        }
+      } else if (strcmp(argv[i], "--log") == 0) {
+        warnNotAndroid("--log");  // Not supported if running on android.
+        if (i + 1 >= argc) {
+          GAPID_FATAL("Usage: --log <log-file-path>");
+        }
+        opts.logPath = argv[++i];
+      } else if (strcmp(argv[i], "--idle-timeout-sec") == 0) {
+        opts.SetMode(kReplayServer);
+        if (i + 1 >= argc) {
+          GAPID_FATAL("Usage: --idle-timeout-sec <timeout in seconds>");
+        }
+        opts.idleTimeoutSec = atoi(argv[++i]);
+      } else if (strcmp(argv[i], "--wait-for-debugger") == 0) {
+        opts.waitForDebugger = true;
+      } else if (strcmp(argv[i], "--version") == 0) {
+        opts.version = true;
+      } else if (strcmp(argv[i], "-h") == 0 || strcmp(argv[i], "-help") == 0 ||
+                 strcmp(argv[i], "--help") == 0) {
+        opts.help = true;
+      } else {
+        GAPID_FATAL("Unknown argument: %s", argv[i]);
+      }
+    }
+
+#if TARGET_OS == GAPID_OS_ANDROID
+    if (opts.android) {
```

Comment:
I am not sure what `--android` is supposed to represent.
It looks like it is supposed to be the equivalent to "--replay-archive" on desktop based on the next little bit of code.

Reply:
--replay-archive does not do the same as --android.

I tried to find a way to express what the old android Parse() function was doing via arguments to the (now unified) PC Parse() function. Unfortunately I didn't see a way of achieving an exact behaviour match without adding a new flag.

Reply:
What about passing a `struct Options *opts` to Parse(), such that you can pre-fill the options before parsing with different defaults depending on whether you're on host or device entry point?

Reply:
Done!